common/components/AddressFieldFactory/AddressFieldDropdown.tsx

```diff
@@ -179,7 +178,7 @@ class AddressFieldDropdownClass extends React.Component<Props, State> {
     const filteredLabelCount = this.getFilteredLabels().length;
 
     let activeIndex =
-      previousActiveIndex === null ? filteredLabelCount - 1 : previousActiveIndex - 1;
+      previousActiveIndex === null ? filteredLabelCount - 1 : previousActiveIndex! - 1;
```

Comment:
It would raise an error complaining about previousActiveIndex was potentially null

Reply:
It was complaining about subtracting from potentially null. So you cast to specify that the value will be defined if this line is executed

Reply:
I see, it can be null. It's better to wrap this in a conditional check than use `!`, which has no runtime checks and just assumes it's non-null even if it is

Reply:
Doesn't the ternary already checks if it's null.
```
previousActiveIndex === null ? filteredLabelCount - 1 : previousActiveIndex - 1
```

More globally I prefer not to change the JS when doing a TS update. If I start refactoring the JS I'm going to be changing the whole component and that never ends.